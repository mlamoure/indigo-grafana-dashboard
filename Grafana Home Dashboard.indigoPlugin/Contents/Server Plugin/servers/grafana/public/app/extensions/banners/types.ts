import React from 'react';

import { BusEventWithPayload } from '@grafana/data';

/** @alpha */
export class PageBannerDisplayEvent extends BusEventWithPayload<PageBannerEventPayload> {
  static type = 'page-banner-display';
}

/** @alpha */
export type PageBannerEventPayload = {
  onClose?: () => void;
  severity?: PageBannerSeverity;
  body: React.ElementType<PageBannerBodyProps>;
};

/** @alpha */
export type PageBannerBodyProps = {
  severity: PageBannerSeverity;
};

/** @alpha */
export type PageBannerAction = {
  text: string;
  href: string;
};

/** @alpha */
export enum PageBannerSeverity {
  info,
  warning,
  error,
}
