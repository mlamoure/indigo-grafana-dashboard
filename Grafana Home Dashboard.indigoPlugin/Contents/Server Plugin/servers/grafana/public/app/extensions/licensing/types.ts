export interface LicenseToken {
  status: number;
  jti: string;
  iss: string;
  sub: string;
  iat: number;
  exp: number;
  nbf: number;
  lexp: number;
  lid: string;
  limit_by: string;
  included_admins: number;
  included_viewers: number;
  included_users: number;
  prod: string[];
  company: string;
  usage_billing?: boolean;
  slug: string;
  details_url?: string;
  account?: string;
  trial_exp?: number;
}

export interface ActiveUserStats {
  active_admins_and_editors: number;
  active_viewers: number;
  active_users: number;
}

export type PermissionsReport = {
  id: number;
  title: string;
  slug: string;
  uid: string;
  url: string;
  isFolder: boolean;
  orgId: number;
  granteeType: string;
  granteeName: string;
  granteeUrl: string;
  customPermissions: string;
  orgRole: string;
  usersCount: number;
};
