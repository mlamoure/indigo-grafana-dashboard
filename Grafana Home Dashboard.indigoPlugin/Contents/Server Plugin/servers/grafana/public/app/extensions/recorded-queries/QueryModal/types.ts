import { DataQuery, DataSourceInstanceSettings } from '@grafana/data';

export interface QueryModalModel {
  title: string;
  body: QueryModalBody;
}

export interface QueryModalBodyProps {
  query?: DataQuery;
  queries?: DataQuery[];
  onAddQuery?: (q: DataQuery) => void;
  onChangeDataSource?: (ds: DataSourceInstanceSettings) => void;
}

export type QueryModalBody = React.ComponentType<QueryModalBodyProps>;
