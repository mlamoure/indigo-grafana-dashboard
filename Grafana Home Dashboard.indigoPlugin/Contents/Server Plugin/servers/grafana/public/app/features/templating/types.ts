import { assignModelProperties } from 'app/core/utils/model_utils';
import { Deferred } from '../../core/utils/deferred';
import { VariableModel as BaseVariableModel } from '@grafana/data';

export enum VariableRefresh {
  never,
  onDashboardLoad,
  onTimeRangeChanged,
}

export enum VariableHide {
  dontHide,
  hideLabel,
  hideVariable,
}

export enum VariableSort {
  disabled,
  alphabeticalAsc,
  alphabeticalDesc,
  numericalAsc,
  numericalDesc,
  alphabeticalCaseInsensitiveAsc,
  alphabeticalCaseInsensitiveDesc,
}

export interface VariableTag {
  selected: boolean;
  text: string | string[];
  values?: any[];
  valuesText?: string;
}

export interface VariableOption {
  selected: boolean;
  text: string | string[];
  value: string | string[];
  isNone?: boolean;
  tags?: VariableTag[];
}

export interface AdHocVariableFilter {
  key: string;
  operator: string;
  value: string;
  condition: string;
}

export interface AdHocVariableModel extends VariableModel {
  datasource: string | null;
  filters: AdHocVariableFilter[];
}

export interface IntervalVariableModel extends VariableWithOptions {
  auto: boolean;
  auto_min: string;
  auto_count: number;
  refresh: VariableRefresh;
}

export interface CustomVariableModel extends VariableWithMultiSupport {}

export interface DataSourceVariableModel extends VariableWithMultiSupport {
  regex: string;
  refresh: VariableRefresh;
}

export interface QueryVariableModel extends DataSourceVariableModel {
  datasource: string | null;
  definition: string;
  sort: VariableSort;
  tags: VariableTag[];
  tagsQuery: string;
  tagValuesQuery: string;
  useTags: boolean;
  queryValue?: string;
}

export interface TextBoxVariableModel extends VariableWithOptions {}

export interface ConstantVariableModel extends VariableWithOptions {}

export interface VariableWithMultiSupport extends VariableWithOptions {
  multi: boolean;
  includeAll: boolean;
  allValue?: string | null;
}

export interface VariableWithOptions extends VariableModel {
  current: VariableOption;
  options: VariableOption[];
  query: string;
}

export interface VariableModel extends BaseVariableModel {
  id?: string; // only exists for variables in redux state
  global?: boolean; // only exists for variables in redux state
  hide: VariableHide;
  skipUrlSync: boolean;
  index?: number;
  initLock?: Deferred | null;
}

export interface VariableActions {
  setValue(option: any): any;
  updateOptions(searchFilter?: string): any;
  dependsOn(variable: any): any;
  setValueFromUrl(urlValue: any): any;
  getValueForUrl(): any;
  getSaveModel(): any;
}

export type CtorType = new (...args: any[]) => {};

export interface VariableTypes {
  [key: string]: {
    name: string;
    ctor: CtorType;
    description: string;
    supportsMulti?: boolean;
  };
}

export let variableTypes: VariableTypes = {};
export { assignModelProperties };
