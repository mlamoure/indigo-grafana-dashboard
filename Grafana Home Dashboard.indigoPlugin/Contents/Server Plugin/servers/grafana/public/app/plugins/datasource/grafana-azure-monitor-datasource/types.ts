import { DataQuery } from '@grafana/ui/src/types';

export interface AzureMonitorQuery extends DataQuery {
  format: string;
  subscription: string;
  azureMonitor: AzureMetricQuery;
  azureLogAnalytics: AzureLogsQuery;
  //   appInsights: any;
}

export interface AzureMetricQuery {
  resourceGroup: string;
  resourceName: string;
  metricDefinition: string;
  metricName: string;
  timeGrainUnit: string;
  timeGrain: string;
  timeGrains: string[];
  aggregation: string;
  dimension: string;
  dimensionFilter: string;
  alias: string;
}

export interface AzureLogsQuery {
  query: string;
  resultFormat: string;
  workspace: string;
}

// Azure Log Analytics types
export interface KustoSchema {
  Databases: { [key: string]: KustoDatabase };
  Plugins: any[];
}
export interface KustoDatabase {
  Name: string;
  Tables: { [key: string]: KustoTable };
  Functions: { [key: string]: KustoFunction };
}

export interface KustoTable {
  Name: string;
  OrderedColumns: KustoColumn[];
}

export interface KustoColumn {
  Name: string;
  Type: string;
}

export interface KustoFunction {
  Name: string;
  DocString: string;
  Body: string;
  Folder: string;
  FunctionKind: string;
  InputParameters: any[];
  OutputColumns: any[];
}

export interface AzureLogsVariable {
  text: string;
  value: string;
}

export interface AzureLogsTableData {
  columns: AzureLogsTableColumn[];
  rows: any[];
  type: string;
  refId: string;
  meta: {
    query: string;
  };
}

export interface AzureLogsTableColumn {
  text: string;
  type: string;
}
