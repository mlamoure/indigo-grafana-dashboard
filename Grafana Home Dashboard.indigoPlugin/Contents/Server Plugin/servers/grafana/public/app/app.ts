import 'babel-polyfill';
import 'file-saver';
import 'lodash';
import 'jquery';
import 'angular';
import 'angular-route';
import 'angular-sanitize';
import 'angular-native-dragdrop';
import 'angular-bindonce';
import 'react';
import 'react-dom';

import 'vendor/bootstrap/bootstrap';
import 'vendor/angular-ui/ui-bootstrap-tpls';
import 'vendor/angular-other/angular-strap';

import $ from 'jquery';
import angular from 'angular';
import config from 'app/core/config';
import _ from 'lodash';
import moment from 'moment';

// add move to lodash for backward compatabiltiy
_.move = (array, fromIndex, toIndex) => {
  array.splice(toIndex, 0, array.splice(fromIndex, 1)[0]);
  return array;
};

import { coreModule, registerAngularDirectives } from './core/core';
import { setupAngularRoutes } from './routes/routes';

declare var System: any;

export class GrafanaApp {
  registerFunctions: any;
  ngModuleDependencies: any[];
  preBootModules: any[];

  constructor() {
    this.preBootModules = [];
    this.registerFunctions = {};
    this.ngModuleDependencies = [];
  }

  useModule(module) {
    if (this.preBootModules) {
      this.preBootModules.push(module);
    } else {
      _.extend(module, this.registerFunctions);
    }
    this.ngModuleDependencies.push(module.name);
    return module;
  }

  init() {
    const app = angular.module('grafana', []);

    moment.locale(config.bootData.user.locale);

    app.config(($locationProvider, $controllerProvider, $compileProvider, $filterProvider, $httpProvider, $provide) => {
      // pre assing bindings before constructor calls
      $compileProvider.preAssignBindingsEnabled(true);

      if (config.buildInfo.env !== 'development') {
        $compileProvider.debugInfoEnabled(false);
      }

      $httpProvider.useApplyAsync(true);

      this.registerFunctions.controller = $controllerProvider.register;
      this.registerFunctions.directive = $compileProvider.directive;
      this.registerFunctions.factory = $provide.factory;
      this.registerFunctions.service = $provide.service;
      this.registerFunctions.filter = $filterProvider.register;

      $provide.decorator('$http', [
        '$delegate',
        '$templateCache',
        ($delegate, $templateCache) => {
          const get = $delegate.get;
          $delegate.get = (url, config) => {
            if (url.match(/\.html$/)) {
              // some template's already exist in the cache
              if (!$templateCache.get(url)) {
                url += '?v=' + new Date().getTime();
              }
            }
            return get(url, config);
          };
          return $delegate;
        },
      ]);
    });

    this.ngModuleDependencies = [
      'grafana.core',
      'ngRoute',
      'ngSanitize',
      '$strap.directives',
      'ang-drag-drop',
      'grafana',
      'pasvaz.bindonce',
      'ui.bootstrap',
      'ui.bootstrap.tpls',
      'react',
    ];

    const moduleTypes = ['controllers', 'directives', 'factories', 'services', 'filters', 'routes'];

    _.each(moduleTypes, type => {
      const moduleName = 'grafana.' + type;
      this.useModule(angular.module(moduleName, []));
    });

    // makes it possible to add dynamic stuff
    this.useModule(coreModule);

    // register react angular wrappers
    coreModule.config(setupAngularRoutes);
    registerAngularDirectives();

    const preBootRequires = [System.import('app/features/all')];

    Promise.all(preBootRequires)
      .then(() => {
        // disable tool tip animation
        $.fn.tooltip.defaults.animation = false;

        // bootstrap the app
        angular.bootstrap(document, this.ngModuleDependencies).invoke(() => {
          _.each(this.preBootModules, module => {
            _.extend(module, this.registerFunctions);
          });

          this.preBootModules = null;
        });
      })
      .catch(err => {
        console.log('Application boot failed:', err);
      });
  }
}

export default new GrafanaApp();
