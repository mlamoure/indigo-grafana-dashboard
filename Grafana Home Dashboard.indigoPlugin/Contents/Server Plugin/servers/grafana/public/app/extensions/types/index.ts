export * from './permissions';
export * from './store';
export * from './reports';
export * from './metaanalytics';
export * from './caching';
export * from './accessControl';
export * from './recordedQuery';
