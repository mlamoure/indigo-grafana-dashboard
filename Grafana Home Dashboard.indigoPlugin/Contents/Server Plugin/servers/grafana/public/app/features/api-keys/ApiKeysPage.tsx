import React, { PureComponent } from 'react';
import ReactDOMServer from 'react-dom/server';
import { connect } from 'react-redux';
import { hot } from 'react-hot-loader';
import { NavModel, ApiKey, NewApiKey, OrgRole } from 'app/types';
import { getNavModel } from 'app/core/selectors/navModel';
import { getApiKeys } from './state/selectors';
import { loadApiKeys, deleteApiKey, setSearchQuery, addApiKey } from './state/actions';
import PageHeader from 'app/core/components/PageHeader/PageHeader';
import SlideDown from 'app/core/components/Animations/SlideDown';
import ApiKeysAddedModal from './ApiKeysAddedModal';
import config from 'app/core/config';
import appEvents from 'app/core/app_events';

export interface Props {
  navModel: NavModel;
  apiKeys: ApiKey[];
  searchQuery: string;
  loadApiKeys: typeof loadApiKeys;
  deleteApiKey: typeof deleteApiKey;
  setSearchQuery: typeof setSearchQuery;
  addApiKey: typeof addApiKey;
}

export interface State {
  isAdding: boolean;
  newApiKey: NewApiKey;
}

enum ApiKeyStateProps {
  Name = 'name',
  Role = 'role',
}

const initialApiKeyState = {
  name: '',
  role: OrgRole.Viewer,
};

export class ApiKeysPage extends PureComponent<Props, any> {
  constructor(props) {
    super(props);
    this.state = { isAdding: false, newApiKey: initialApiKeyState };
  }

  componentDidMount() {
    this.fetchApiKeys();
  }

  async fetchApiKeys() {
    await this.props.loadApiKeys();
  }

  onDeleteApiKey(key: ApiKey) {
    this.props.deleteApiKey(key.id);
  }

  onSearchQueryChange = evt => {
    this.props.setSearchQuery(evt.target.value);
  };

  onToggleAdding = () => {
    this.setState({ isAdding: !this.state.isAdding });
  };

  onAddApiKey = async evt => {
    evt.preventDefault();

    const openModal = (apiKey: string) => {
      const rootPath = window.location.origin + config.appSubUrl;
      const modalTemplate = ReactDOMServer.renderToString(<ApiKeysAddedModal apiKey={apiKey} rootPath={rootPath} />);

      appEvents.emit('show-modal', {
        templateHtml: modalTemplate,
      });
    };

    this.props.addApiKey(this.state.newApiKey, openModal);
    this.setState((prevState: State) => {
      return {
        ...prevState,
        newApiKey: initialApiKeyState,
      };
    });
  };

  onApiKeyStateUpdate = (evt, prop: string) => {
    const value = evt.currentTarget.value;
    this.setState((prevState: State) => {
      const newApiKey = {
        ...prevState.newApiKey,
      };
      newApiKey[prop] = value;

      return {
        ...prevState,
        newApiKey: newApiKey,
      };
    });
  };

  render() {
    const { newApiKey, isAdding } = this.state;
    const { navModel, apiKeys, searchQuery } = this.props;

    return (
      <div>
        <PageHeader model={navModel} />
        <div className="page-container page-body">
          <div className="page-action-bar">
            <div className="gf-form gf-form--grow">
              <label className="gf-form--has-input-icon gf-form--grow">
                <input
                  type="text"
                  className="gf-form-input"
                  placeholder="Search keys"
                  value={searchQuery}
                  onChange={this.onSearchQueryChange}
                />
                <i className="gf-form-input-icon fa fa-search" />
              </label>
            </div>

            <div className="page-action-bar__spacer" />
            <button className="btn btn-success pull-right" onClick={this.onToggleAdding} disabled={isAdding}>
              <i className="fa fa-plus" /> Add API Key
            </button>
          </div>

          <SlideDown in={isAdding}>
            <div className="cta-form">
              <button className="cta-form__close btn btn-transparent" onClick={this.onToggleAdding}>
                <i className="fa fa-close" />
              </button>
              <h5>Add API Key</h5>
              <form className="gf-form-group" onSubmit={this.onAddApiKey}>
                <div className="gf-form-inline">
                  <div className="gf-form max-width-21">
                    <span className="gf-form-label">Key name</span>
                    <input
                      type="text"
                      className="gf-form-input"
                      value={newApiKey.name}
                      placeholder="Name"
                      onChange={evt => this.onApiKeyStateUpdate(evt, ApiKeyStateProps.Name)}
                    />
                  </div>
                  <div className="gf-form">
                    <span className="gf-form-label">Role</span>
                    <span className="gf-form-select-wrapper">
                      <select
                        className="gf-form-input gf-size-auto"
                        value={newApiKey.role}
                        onChange={evt => this.onApiKeyStateUpdate(evt, ApiKeyStateProps.Role)}
                      >
                        {Object.keys(OrgRole).map(role => {
                          return (
                            <option key={role} label={role} value={role}>
                              {role}
                            </option>
                          );
                        })}
                      </select>
                    </span>
                  </div>
                  <div className="gf-form">
                    <button className="btn gf-form-btn btn-success">Add</button>
                  </div>
                </div>
              </form>
            </div>
          </SlideDown>

          <h3 className="page-heading">Existing Keys</h3>
          <table className="filter-table">
            <thead>
              <tr>
                <th>Name</th>
                <th>Role</th>
                <th style={{ width: '34px' }} />
              </tr>
            </thead>
            {apiKeys.length > 0 ? (
              <tbody>
                {apiKeys.map(key => {
                  return (
                    <tr key={key.id}>
                      <td>{key.name}</td>
                      <td>{key.role}</td>
                      <td>
                        <a onClick={() => this.onDeleteApiKey(key)} className="btn btn-danger btn-mini">
                          <i className="fa fa-remove" />
                        </a>
                      </td>
                    </tr>
                  );
                })}
              </tbody>
            ) : null}
          </table>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    navModel: getNavModel(state.navIndex, 'apikeys'),
    apiKeys: getApiKeys(state.apiKeys),
    searchQuery: state.apiKeys.searchQuery,
  };
}

const mapDispatchToProps = {
  loadApiKeys,
  deleteApiKey,
  setSearchQuery,
  addApiKey,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(ApiKeysPage));
