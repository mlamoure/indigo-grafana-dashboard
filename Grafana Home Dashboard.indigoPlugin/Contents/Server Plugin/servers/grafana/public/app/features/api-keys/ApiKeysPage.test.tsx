import React from 'react';
import { shallow } from 'enzyme';
import { Props, ApiKeysPage } from './ApiKeysPage';
import { NavModel, ApiKey } from 'app/types';
import { getMultipleMockKeys, getMockKey } from './__mocks__/apiKeysMock';

const setup = (propOverrides?: object) => {
  const props: Props = {
    navModel: {} as NavModel,
    apiKeys: [] as ApiKey[],
    searchQuery: '',
    loadApiKeys: jest.fn(),
    deleteApiKey: jest.fn(),
    setSearchQuery: jest.fn(),
    addApiKey: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<ApiKeysPage {...props} />);
  const instance = wrapper.instance() as ApiKeysPage;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render API keys table', () => {
    const { wrapper } = setup({
      apiKeys: getMultipleMockKeys(5),
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Life cycle', () => {
  it('should call loadApiKeys', () => {
    const { instance } = setup();

    instance.componentDidMount();

    expect(instance.props.loadApiKeys).toHaveBeenCalled();
  });
});

describe('Functions', () => {
  describe('Delete team', () => {
    it('should call delete team', () => {
      const { instance } = setup();
      instance.onDeleteApiKey(getMockKey());
      expect(instance.props.deleteApiKey).toHaveBeenCalledWith(1);
    });
  });

  describe('on search query change', () => {
    it('should call setSearchQuery', () => {
      const { instance } = setup();
      const mockEvent = { target: { value: 'test' } };

      instance.onSearchQueryChange(mockEvent);

      expect(instance.props.setSearchQuery).toHaveBeenCalledWith('test');
    });
  });
});
