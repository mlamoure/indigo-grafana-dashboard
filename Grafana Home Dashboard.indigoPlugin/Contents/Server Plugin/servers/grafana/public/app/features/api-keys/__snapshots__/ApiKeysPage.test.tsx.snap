// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render API keys table 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="page-action-bar"
    >
      <div
        className="gf-form gf-form--grow"
      >
        <label
          className="gf-form--has-input-icon gf-form--grow"
        >
          <input
            className="gf-form-input"
            onChange={[Function]}
            placeholder="Search keys"
            type="text"
            value=""
          />
          <i
            className="gf-form-input-icon fa fa-search"
          />
        </label>
      </div>
      <div
        className="page-action-bar__spacer"
      />
      <button
        className="btn btn-success pull-right"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-plus"
        />
         Add API Key
      </button>
    </div>
    <Component
      in={false}
    >
      <div
        className="cta-form"
      >
        <button
          className="cta-form__close btn btn-transparent"
          onClick={[Function]}
        >
          <i
            className="fa fa-close"
          />
        </button>
        <h5>
          Add API Key
        </h5>
        <form
          className="gf-form-group"
          onSubmit={[Function]}
        >
          <div
            className="gf-form-inline"
          >
            <div
              className="gf-form max-width-21"
            >
              <span
                className="gf-form-label"
              >
                Key name
              </span>
              <input
                className="gf-form-input"
                onChange={[Function]}
                placeholder="Name"
                type="text"
                value=""
              />
            </div>
            <div
              className="gf-form"
            >
              <span
                className="gf-form-label"
              >
                Role
              </span>
              <span
                className="gf-form-select-wrapper"
              >
                <select
                  className="gf-form-input gf-size-auto"
                  onChange={[Function]}
                  value="Viewer"
                >
                  <option
                    key="Viewer"
                    label="Viewer"
                    value="Viewer"
                  >
                    Viewer
                  </option>
                  <option
                    key="Editor"
                    label="Editor"
                    value="Editor"
                  >
                    Editor
                  </option>
                  <option
                    key="Admin"
                    label="Admin"
                    value="Admin"
                  >
                    Admin
                  </option>
                </select>
              </span>
            </div>
            <div
              className="gf-form"
            >
              <button
                className="btn gf-form-btn btn-success"
              >
                Add
              </button>
            </div>
          </div>
        </form>
      </div>
    </Component>
    <h3
      className="page-heading"
    >
      Existing Keys
    </h3>
    <table
      className="filter-table"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Role
          </th>
          <th
            style={
              Object {
                "width": "34px",
              }
            }
          />
        </tr>
      </thead>
      <tbody>
        <tr
          key="1"
        >
          <td>
            test-1
          </td>
          <td>
            Viewer
          </td>
          <td>
            <a
              className="btn btn-danger btn-mini"
              onClick={[Function]}
            >
              <i
                className="fa fa-remove"
              />
            </a>
          </td>
        </tr>
        <tr
          key="2"
        >
          <td>
            test-2
          </td>
          <td>
            Viewer
          </td>
          <td>
            <a
              className="btn btn-danger btn-mini"
              onClick={[Function]}
            >
              <i
                className="fa fa-remove"
              />
            </a>
          </td>
        </tr>
        <tr
          key="3"
        >
          <td>
            test-3
          </td>
          <td>
            Viewer
          </td>
          <td>
            <a
              className="btn btn-danger btn-mini"
              onClick={[Function]}
            >
              <i
                className="fa fa-remove"
              />
            </a>
          </td>
        </tr>
        <tr
          key="4"
        >
          <td>
            test-4
          </td>
          <td>
            Viewer
          </td>
          <td>
            <a
              className="btn btn-danger btn-mini"
              onClick={[Function]}
            >
              <i
                className="fa fa-remove"
              />
            </a>
          </td>
        </tr>
        <tr
          key="5"
        >
          <td>
            test-5
          </td>
          <td>
            Viewer
          </td>
          <td>
            <a
              className="btn btn-danger btn-mini"
              onClick={[Function]}
            >
              <i
                className="fa fa-remove"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Render should render component 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="page-action-bar"
    >
      <div
        className="gf-form gf-form--grow"
      >
        <label
          className="gf-form--has-input-icon gf-form--grow"
        >
          <input
            className="gf-form-input"
            onChange={[Function]}
            placeholder="Search keys"
            type="text"
            value=""
          />
          <i
            className="gf-form-input-icon fa fa-search"
          />
        </label>
      </div>
      <div
        className="page-action-bar__spacer"
      />
      <button
        className="btn btn-success pull-right"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa fa-plus"
        />
         Add API Key
      </button>
    </div>
    <Component
      in={false}
    >
      <div
        className="cta-form"
      >
        <button
          className="cta-form__close btn btn-transparent"
          onClick={[Function]}
        >
          <i
            className="fa fa-close"
          />
        </button>
        <h5>
          Add API Key
        </h5>
        <form
          className="gf-form-group"
          onSubmit={[Function]}
        >
          <div
            className="gf-form-inline"
          >
            <div
              className="gf-form max-width-21"
            >
              <span
                className="gf-form-label"
              >
                Key name
              </span>
              <input
                className="gf-form-input"
                onChange={[Function]}
                placeholder="Name"
                type="text"
                value=""
              />
            </div>
            <div
              className="gf-form"
            >
              <span
                className="gf-form-label"
              >
                Role
              </span>
              <span
                className="gf-form-select-wrapper"
              >
                <select
                  className="gf-form-input gf-size-auto"
                  onChange={[Function]}
                  value="Viewer"
                >
                  <option
                    key="Viewer"
                    label="Viewer"
                    value="Viewer"
                  >
                    Viewer
                  </option>
                  <option
                    key="Editor"
                    label="Editor"
                    value="Editor"
                  >
                    Editor
                  </option>
                  <option
                    key="Admin"
                    label="Admin"
                    value="Admin"
                  >
                    Admin
                  </option>
                </select>
              </span>
            </div>
            <div
              className="gf-form"
            >
              <button
                className="btn gf-form-btn btn-success"
              >
                Add
              </button>
            </div>
          </div>
        </form>
      </div>
    </Component>
    <h3
      className="page-heading"
    >
      Existing Keys
    </h3>
    <table
      className="filter-table"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Role
          </th>
          <th
            style={
              Object {
                "width": "34px",
              }
            }
          />
        </tr>
      </thead>
    </table>
  </div>
</div>
`;
